app/src/main/java/ru/vssemikoz/newsfeed/MainActivity.java

```diff
@@ -5,34 +5,65 @@
 import androidx.recyclerview.widget.RecyclerView;
 
 import android.os.Bundle;
+import android.util.Log;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
 import ru.vssemikoz.newsfeed.adapters.NewsFeedAdapter;
+import ru.vssemikoz.newsfeed.api.NewsApi;
 import ru.vssemikoz.newsfeed.models.NewsItem;
+import ru.vssemikoz.newsfeed.models.NewsItemList;
 
 public class MainActivity extends AppCompatActivity {
-
+    String KEY = "c94a57cbbb50497f94a2bb167dc91fc5";
     List<NewsItem> newsItems = new ArrayList<NewsItem>();
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        initData();
+        Callback<NewsItemList> callbackNewsItemList = new Callback<NewsItemList>() {
```

Comment:
Looks cool. But we should not keep such things in `onCreate()`. We should at least create another method and call it from `onCreate()`. Let's do it ;)